### PR TITLE
Delete the sysctl runtime admit handler

### PR DESCRIPTION
**Pull request by mattjmcnaughton**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
As of https://github.com/kubernetes/kubernetes/pull/72831, the minimum
docker version is 1.13.1. (and the minimum API version is 1.26). The
only time the `RuntimeAdmitHandler` returns anything other than accept
is when the Docker API version < 1.24. In other words, we can be
confident that Docker will always support sysctl.

As a result, we can delete this unnecessary and docker-specific code.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs
NONE
```

### CODE REVIEW

**Comment by mattjmcnaughton** [comment]
cc @yujuhong as the author of the original pr.
cc @dims (another instance of cleaning up docker-specific code in the kubelet).

**Comment by mattjmcnaughton** [comment]
On Tue, Jan 21, 2020 at 11:16:52AM -0800, Kubernetes Prow Robot wrote:
> @mattjmcnaughton: The following test **failed**, say `/retest` to rerun all failed tests:
> 
> Test name | Commit | Details | Rerun command
> --- | --- | --- | ---
> pull-kubernetes-e2e-gce | 61c172f67808dfb92023571850f23b81bb861b24 | [link](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/87394/pull-kubernetes-e2e-gce/1219684562875977732/) | `/test pull-kubernetes-e2e-gce`
> 
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes&repo=kubernetes&pr=87394). [Your PR dashboard](https://gubernator.k8s.io/pr/mattjmcnaughton). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open) when you hit one in your PR.
> 
> <details>
> 
> Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
> </details>
> 
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/kubernetes/kubernetes/pull/87394#issuecomment-576835985

/retest

**Comment by mattjmcnaughton** [comment]
/assign @smarterclayton

@yujuhong mind re-marking lgtm after fixing merge conflict? thanks :)

**Comment by mattjmcnaughton** [comment]
/assign @liggitt 

cc @yujuhong mind adding back your approval? I had to rebase - thanks :)

File: test/e2e/framework/pods.go [hunk 1]
```diff
@@ -226,7 +226,7 @@ func (c *PodClient) WaitForErrorEventOrSuccess(pod *v1.Pod) (*v1.Event, error) {
 		}
 		for _, e := range evnts.Items {
 			switch e.Reason {
-			case events.KillingContainer, events.FailedToCreateContainer, sysctl.UnsupportedReason, sysctl.ForbiddenReason:
```
Comment (liggitt) [inline test/e2e/framework/pods.go]: does this test run against skewed kubelets that would still return this error?

**Comment by mattjmcnaughton** [comment]
On Tue, Jan 28, 2020 at 06:13:56AM -0800, Jordan Liggitt wrote:
> liggitt commented on this pull request.
> 
> 
> 
> > @@ -226,7 +226,7 @@ func (c *PodClient) WaitForErrorEventOrSuccess(pod *v1.Pod) (*v1.Event, error) {
>  		}
>  		for _, e := range evnts.Items {
>  			switch e.Reason {
> -			case events.KillingContainer, events.FailedToCreateContainer, sysctl.UnsupportedReason, sysctl.ForbiddenReason:
> 
> does this test run against skewed kubelets that would still return this error?
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/kubernetes/kubernetes/pull/87394#pullrequestreview-349403925

Great question! Do the skewed tests run as part of CI? If not, is there a way I
can run them and view the results?

**Comment by liggitt** [comment]
they run post-merge, as far as I know :-/

**Comment by mattjmcnaughton** [comment]
> they run post-merge, as far as I know :-/

@liggitt I've been thinking about this and I believe we should be fine. #72831 was merged in Jan 2019, which means it was definitely in 1.15 (oldest supported minor version). Unless, do we run skew tests with non-supported versions?

Thanks for your help here :) To be honest, I don't know a ton about the current skew testing set up.

**Comment by liggitt** [comment]
ok, thanks for digging
/lgtm
/approve

**Comment by mattjmcnaughton** [comment]
On Thu, Jan 30, 2020 at 07:22:06PM -0800, Jordan Liggitt wrote:
> ok, thanks for digging
> /lgtm
> /approve
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/kubernetes/kubernetes/pull/87394#issuecomment-580564482

Thank you, @liggitt!

**Pull request merged**